### PR TITLE
Feature flags v1

### PR DESCRIPTION
**Pull request by AntonEbel**
close Graylog2/graylog-plugin-cloud#938

- [X] New feature (non-breaking change which adds functionality)
- [ ] The commit history must be preserved - please use the rebase-merge or standard merge option instead of squash-merge
- [ ] Sync up with the author before merging

/jenkins-pr-deps Graylog2/graylog-plugin-cloud#952

### CODE REVIEW

**Review by patrickmann — COMMENTED** [review]
Is there a test that reads a config file of flag?

**Review by patrickmann — APPROVED** [review]
Looks good

File: graylog2-server/src/main/java/org/graylog2/featureflag/ImmutableFeatureFlags.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+                .collect(Collectors.toMap(e -> toUpperCase(e.getKey()), e -> new FeatureFlag(e.getKey(), e.getValue(), metricRegistry)));
+    }
+
+    @Override
+    public Map<String, String> getAll() {
+        return flags.values().stream().collect(Collectors.toMap(flag -> flag.name, flag -> flag.value));
+    }
+
+    @Override
+    public boolean isOn(String feature, boolean defaultValue) {
```
Comment (bernd) [inline graylog2-server/src/main/java/org/graylog2/featureflag/ImmutableFeatureFlags.java:48]: I was wondering what the use case for the `defaultValue` is. According to the pitch a feature flag should be off by default, when would one set the `defaultValue` to true?

Comment (AntonEbel) [reply graylog2-server/src/main/java/org/graylog2/featureflag/ImmutableFeatureFlags.java:48]: The purpose of the defaultValue was to give the developer some flexibility and to make clear what the result would be if the feature is not present. But you are right I can't think of any use case where I would set the default value to true. Maybe it's not worth it to have this flexibility. If we remove the default parameter we should provide at least some JavaDoc to make clear what is the fallback behavior or come up with good name for the method.

Comment (bernd) [reply graylog2-server/src/main/java/org/graylog2/featureflag/ImmutableFeatureFlags.java:48]: Sounds good! We can always add a method with a default value later if we see that we need it. Taking the parameter away is way harder once the API is in use. :+1:

**Review by patrickmann — APPROVED** [review]

File: graylog2-server/src/main/java/org/graylog2/featureflag/ImmutableFeatureFlags.java (hunk 1 above)
Comment (AntonEbel) [reply graylog2-server/src/main/java/org/graylog2/featureflag/ImmutableFeatureFlags.java]: I removed the defaultValue parameter.

**Pull request merged**